### ISSUE TITLE
Unfriendly Simbad.query_criteria functionality

### ISSUE THREAD

**Issue by DinoBektesevic** [astropy/astroquery#2041]
I either can't figure out how to, or the functionality for it isn't there but using `Simbad.query_criteria` wants me to pass full strings of my queries and will not accept SkyCoords. Currently [the documentation ](https://astroquery.readthedocs.io/en/latest/simbad/simbad.html#query-based-on-any-criteria) shows these two usage cases:

```
result = Simbad.query_criteria('region(box, GAL, 0 +0, 3d 1d)', otype='SNR')
...
script = '(region(box, GAL, 0 +0, 0.5d 0.5d) | region(box, GAL, 43.3 -0.2, 0.25d 0.25d))'
result = Simbad.query_criteria(script, otype='SNR')
```

Could it be possible to add some friendliness to how this is scripted for some of the easier criterion operators such as `region`? For example:
```
result = Simbad.query_criteria(SkyCoord, criterion="region", shape="circle", radius=Unit)
```

Or something similar?

**Comment by bsipocz** [comment]
My thinking would be to provide a way to add criteria to the usual API methods, e.g. 
```
query_region(coord, criteria={'otype': 'STAR',}
```

 to query for stars, rather than forcing you to use `query_criteria`, and define regions by hand, etc. At the end of the day, we query the same API so this will all boil down to creating payload within astroquery.

**Comment by ManonMarchand** [comment]
Hi there, 

I'm working on query criteria now and would like to pick your minds. 

I like a lot brigitta's idea but what about different operators than `=`? (ADQL has =, !=, <>, <, >, <=, >=, BETWEN, IN, LIKE, IS NULL, IS NOT NULL, EXISTS')  And what about other logical operators than `AND` (namely `OR` and `NOT` are also possible there)? 

I was thinking to introduce a Criteria dataclass (or named tuples I don't know what is more is astroquery's philosophy). Maybe something along: 

```python
from dataclasses import dataclass
from typing import Any

@dataclass
class Criteria:
    column: str
    operator: str
    value: Any
    logic = 'AND'

Criteria('ra', '<=', '10')
```

```
Criteria(column='ra', value='<=', operator='10', logic='AND')
```

Then it would be called like this : 

```python
query_region(coord, Criteria('otype', '!=', 'Cl*'))
```

(where the signature of query_region would be `query_region(coord, **criteria)`)

**Comment by bsipocz** [comment]
> I like a lot brigitta's idea but what about different operators than =? (ADQL has =, !=, <>, <, >, <=, >=, BETWEN, IN, LIKE, IS NULL, IS NOT NULL, EXISTS') And what about other logical operators than AND (namely OR and NOT are also possible there)?

I wonder how much of this scenario gets resolved with #2856 getting merged, practically asking users to write ADQL queries for the more complicated ones.

**Comment by ManonMarchand** [comment]
This is a bit different from adding the possibility to do tap queries. We intend to deprecate the sim-script API, so the next contribution from cds will be to rewrite the existing Simbadclass query methods to use TAP (issue #1468). I'm wondering if it could be an occasion to add this feature that I find quite useful. It'd be a way to do moderately intricate queries (a few joins / criteria only) for the non adql-proficient users
